sdks/router-sdk/src/entities/mixedRoute/route.ts

```diff
@@ -54,17 +55,22 @@ export class MixedRouteSDK<TInput extends Currency, TOutput extends Currency> {
       const prevPool = pools[i - 1]
       const pool = pools[i]
       const inputToken = tokenPath[i]
+      const isWrappedNativeEquality = ((pool.token0.equals(inputToken.wrapped) || pool.token1.equals(inputToken.wrapped)) && inputToken.isNative)
```

Comment:
so this is just checking if:
inputToken is native AND token0 or token1 are wrappedNative OR
inputToken is wrappedNative AND token0 or token1 are native

ie these two cases:
path token: ETH & pool: (WETH-X) or
path token: WETH & pool (ETH-X)

I think this works but I do wonder if you could also just check if the inputToken != either token0 or token1.. probably not as future proof but would make this logic a little easier to reason through